app/logic/userManagement.py

```diff
@@ -53,8 +72,11 @@ def addNextTerm():
     newTerm = Term.create(
             description=newDescription,
             year=newYear,
-            academicYear=newAY, 
+            academicYear=newAY,
             isSummer="Summer" in newDescription.split())
     newTerm.save()
 
     return newTerm
+
+def getPrograms():
+    return Program.select().join(StudentManager).where(StudentManager.user==g.current_user).order_by(Program.programName)
```

Comment:
This should take a passed-in user, rather than use g.current_user

Reply:
Fixed!